engine/src/main/java/com/datatorrent/stram/stream/OiOStream.java

```diff
@@ -29,11 +29,12 @@
  *
  * @since 0.3.5
  */
-public class OiOStream implements Stream, SweepableReservoir
```

Comment:
It is necessary to maintain two different counts - one for Sink and one for SweepableReservoir as they are reset separately. In case, OioStream implements both Stream and SweepableReservoir, it will not be possible to distinguish which count should be reset to 0, so either `Sink.getCount()` or `SweepableReservoir.getCount()` would return a wrong result.